### PR TITLE
Consolidating code around process exiting

### LINKED ISSUE #13748: Use proper process exiting mechanism in FSC
On the review with @KevinRansom, we've realized that there's an existing mechanism for proper exit from the compiler - using of exiters (with StopProcessing exceptions probably).

For example:
https://github.com/dotnet/fsharp/blob/af0015e5cac978bef2aa10264f63c886a2532bda/src/fsc/fscmain.fs#L77-L108

or:

https://github.com/dotnet/fsharp/blob/af0015e5cac978bef2aa10264f63c886a2532bda/src/Compiler/Service/service.fs#L119-L129

Wondering how easy will it be to plug into the cli options, since it is designed for cases like this and is probably the right way of dealing with it.

_Originally posted by @vzarytovskii in https://github.com/dotnet/fsharp/issues/13702#issuecomment-1222189416_

### PR DESCRIPTION
**Pull request by psfinaki**
closes #13748 

So apparently we already have the right thing in the codebase so just plugging it in in a couple places.
QuitProcessExiter is [here](https://github.com/dotnet/fsharp/blob/8385830e1f36e5005f952bd1267b969922d0b20f/src/Compiler/Facilities/DiagnosticsLogger.fs#L167).

### CODE REVIEW

**Review by 0101 — APPROVED** [review]

**Review by KevinRansom — CHANGES_REQUESTED** [review]
I think we should consider parameterizing the process exiter.

File: src/Compiler/Driver/ParseAndCheckInputs.fs [hunk 1]
```diff
@@ -737,7 +737,7 @@ let ParseOneInputFile (tcConfig: TcConfig, lexResourceManager, fileName, isLastC
 /// Parse multiple input files from disk
 let ParseInputFiles
     (
-        tcConfig: TcConfig,
+        tcConfigB: TcConfigBuilder,
```
Comment (KevinRansom) [inline src/Compiler/Driver/ParseAndCheckInputs.fs:740]: probably not this.

**Review by KevinRansom — APPROVED** [review]
Nice.

Thank you

**Review by vzarytovskii — APPROVED** [review]

**Pull request merged**